packages/web-config-server/src/apiV1/utils/findAccessibleGroupedMapOverlays.js

```diff
@@ -93,14 +93,21 @@ const checkIfGroupedMapOverlaysAreEmpty = nestedMapOverlayGroups => {
   });
 };
 
-const allRelationsHaveSortOrder = relations => relations.every(m => m.sort_order !== null);
-
-const getSortFunction = relations => {
+/**
+ * Sort map overlays/map overlay groups with sort_order first, then the ones without sort_order alphabetically
+ * @param {*} mapOverlayItems 
+ * @param {*} relations 
+ */
+const sortMapOverlayItems = (mapOverlayItems, relations) => {
   const childIdToSortOrder = reduceToDictionary(relations, 'child_id', 'sort_order');
-  const sortBySortOrder = (m1, m2) => childIdToSortOrder[m1.id] - childIdToSortOrder[m2.id];
-
-  //If all the relations have sort_order, sort by the sort_order. Otherwise by default sort by name.
-  return allRelationsHaveSortOrder(relations) ? sortBySortOrder : getSortByKey('name');
+  const sortedOverlaysByOrder = mapOverlayItems
+    .filter(m => childIdToSortOrder[m.id] !== null)
+    .sort((m1, m2) => childIdToSortOrder[m1.id] - childIdToSortOrder[m2.id]);
+  const sortedOverlaysAlphabetically = mapOverlayItems
+    .filter(m => childIdToSortOrder[m.id] === null)
+    .sort(getSortByKey('name'));
+
+  return [...sortedOverlaysByOrder, ...sortedOverlaysAlphabetically];
```

Comment:
Before if any of the items in a list of map overlays or map overlay groups doesn't have a sort_order, the whole list will be sorted alphabetically. I've modified this to sort the ones that have sort_order first to allow more flexibility